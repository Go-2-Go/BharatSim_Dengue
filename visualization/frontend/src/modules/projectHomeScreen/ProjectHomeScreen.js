import React, { useContext } from 'react';
import { useHistory } from 'react-router-dom';
import { Box, Typography } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import ClickableCard from '../../uiComponent/ClickableCard';
import useModal from '../../hook/useModal';
import CreateNewDashboardModal from './CreateNewDashboardModal';
import LoaderOrError from '../../component/loaderOrError/LoaderOrError';
import useFetchExecutor from '../../hook/useFetchExecuter';
import { api } from '../../utils/api';
import ProjectHeader from '../../uiComponent/ProjectHeader';
import { projectLayoutContext } from '../../contexts/projectLayoutContext';

const useStyles = makeStyles((theme) => {
  return {
    addProjectContainer: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      flexDirection: 'column',
      height: theme.spacing(66),
      boxSizing: 'border-box',
    },
  };
});

function ProjectHomeScreen() {
  const classes = useStyles();
  const history = useHistory();
  const { executeFetch, loadingState } = useFetchExecutor();
  const { openModal, isOpen, closeModal } = useModal();
  const { projectMetadata } = useContext(projectLayoutContext);

  async function onCreate(values) {
    closeModal();
    const { projectId } = await executeFetch(api.saveProject, [{ name: values['project-title'] }]);
    await executeFetch(api.addNewDashboard, [{ name: values['dashboard-title'], projectId }]);
    history.push(`/projects/${projectId}`);
  }

  return (
    <LoaderOrError loadingState={loadingState}>
      <Box>
        <ProjectHeader>{projectMetadata.name}</ProjectHeader>
        <Box py={14} px={32}>
          <ClickableCard onClick={openModal}>
            <Box className={classes.addProjectContainer}>
              <Box pb={2}>
                <Typography variant="h6"> You donâ€™t have any dashboards Yet. </Typography>
              </Box>
              <Typography variant="body2"> Click here to create your first dashboard. </Typography>
            </Box>
          </ClickableCard>
        </Box>
        <CreateNewDashboardModal isOpen={isOpen} closeModal={closeModal} onCreate={onCreate} />
      </Box>
    </LoaderOrError>
  );
}

export default ProjectHomeScreen;
