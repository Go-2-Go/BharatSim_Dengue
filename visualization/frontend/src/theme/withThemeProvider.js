import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme() {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
