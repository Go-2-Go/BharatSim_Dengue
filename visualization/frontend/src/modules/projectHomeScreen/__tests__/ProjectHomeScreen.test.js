import React from 'react';
import { render, within } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import ProjectHomeScreen from '../ProjectHomeScreen';
import withThemeProvider from '../../../theme/withThemeProvider';

describe('<ProjectHomeScreenComponent />', () => {
  const ProjectHomeScreenComponent = withThemeProvider(ProjectHomeScreen);
  it('should match snapshot', () => {
    const { container } = render(<ProjectHomeScreenComponent />);
    expect(container).toMatchSnapshot();
  });
  it('should open new dashboard popup onclick of card', () => {
    const { getByText } = render(<ProjectHomeScreenComponent />);
    fireEvent.click(getByText('Click here to create your first dashboard.'));

    const container = within(document.querySelector('.MuiPaper-root'));

    expect(container.queryByText('New Dashboard')).toBeInTheDocument();
  });
});
