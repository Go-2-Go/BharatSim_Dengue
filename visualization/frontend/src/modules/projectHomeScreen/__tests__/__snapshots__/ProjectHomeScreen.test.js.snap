// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClickableCard /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <div
      class="MuiBox-root MuiBox-root-4 makeStyles-ClickableCardContainer-3"
    >
      <div
        class="MuiBox-root MuiBox-root-5 makeStyles-addProjectContainer-1"
      >
        <div
          class="MuiBox-root MuiBox-root-6"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
             You donâ€™t have any dashboards Yet. 
          </h6>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
           Click here to create your first dashboard. 
        </p>
      </div>
    </div>
  </div>
</div>
`;
